extract-requires.py

```diff
@@ -70,6 +71,19 @@ def get_build_backend_hook_caller(pyproject_toml):
     )
 
 
+def evaluate_marker(req, extras=None):
+    marker_env = markers.default_environment()
+    for extra in extras if extras else [""]:
```

Comment:
I'm used to seeing something like `for extra in (extras or [])` to ensure that the list iterates over a valid iterable. The subtlety of needing a non-empty list threw me at first.

Reply:
Thanks, yeah - I didn't like it either. I pushed a re-work in commit 610073e